src/main/scala/rules/Executor.scala

```diff
@@ -46,7 +47,7 @@ object executor extends ExecutionRules {
   import evaluator._
   import producer._
 
-  private def follow(s: State, edge: SilverEdge, v: Verifier)
+  private def follow(s: State, edge: SilverEdge, v: Verifier, joinPoint: Option[SilverBlock])
```

Comment:
Maybe worthwhile: if you define the last argument to be `joinPoint: Option[SilverBlock] = None`, then no argument has to be explicitly provided at call site in the `None` case.

Reply:
That is what I've done originally, but then I discovered a bug where I forgot to supply `joinPoint`, and it was automatically assumed to be `None`. I omitted default initialization with `None` to avoid such bugs in the future, so one has to explicitly use `joinPoint = None`. This makes it more clear in my opinion, but I can change it if you prefer otherwise.

Reply:
Nope, all good